tools/build-webview

```diff
@@ -29,119 +37,169 @@ err() {
+        ios)
+            if [ "$(uname)" != "Darwin" ]; then
+                err "iOS builds only supported on macOS";
+            fi
+
+            # $dest should be copied to `${bundleroot}/webview`.
+            #
+            # $dest itself must be specified twice in Xcode: once in the invocation
+            # of this script, and once in the "Copy Bundle Resources" step. If you
+            # change it, you'll need to change it in both places. (And here, of
+            # course.)
+            if [[ "$dest" != "$root/ios/"* ]]; then
+                err "unexpected destination directory '$dest' (expected target in iOS build dir)"
+            fi
+        ;;
+        android)
+            # $dest should be copied to `file:///android_asset/webview`.
+            #
+            # Gradle -- more precisely, the Android Gradle plugin -- is already
+            # configured to copy some directory from `build/` into the APK. We
+            # determine which directory that is in Gradle, and pass it here.
+            if [[ "$dest" != "$root/android/app/build/"* ]]; then
+                err "unexpected destination directory '$dest' (expected target in Android build dir)"
+            fi
+        ;;
+        *) err "impossible target $target";;
+    esac
 
-# Set SRC.
-SRC=src/webview/static
-if [ ! -d "$SRC" ]; then
-    err "cannot find asset source directory '$SRC'";
-fi
+    # As implied above, the ultimate destination of these files should be a
+    # directory named `./webview` (relative to the platform-dependent asset
+    # root directory).
+    #
+    # The target directory basename should therefore be 'webview': both build
+    # systems will preserve the name they're given here.
+    if [[ "$dest" != *"/webview" ]]; then
+        err "unexpected destination directory '$dest'"
+    fi
+fi  # if sanity_checks
 
 
 ################################################################################
-# Pre-sync: platform-agnostic build steps
+# Pre-sync: additional build steps
 ################################################################################
 
-# Currently none, but this is where they'll go.
+# Move to the project root.
 #
+# (Ideally, neither this script nor any of its callees should depend on the CWD
+# being anything in particular, but it's an easy thing to depend on without
+# realizing it.)
+cd "$root"
+
 # TODO: Use something like `make -f src/webview/static.Makefile`, to ensure that
 #   complicated build steps are not run unconditionally.
 #
 # TODO: split assets into build-time-generated and build-time-static?
 #   - rsync can sync from multiple directories, but has no collision-detection
 #   - might be too many layers of abstraction
 
+# Run each pre-sync auxiliary script.
+#
+# Each script gets a few selected parameters as environment variables.
+for script in "${aux_script_dir}"/*.{all,"${target}"}.pre; do
+    [ -f "$script" ] || continue
+    ROOT="$root" TARGET="$target" STAGING="$staging" "$script"
```

Comment:
formatting nit: having the main verb way off on the right makes it more work to parse this than usual. A simple fix is
```
    ROOT="$root" TARGET="$target" STAGING="$staging" \
      "$script"
```